### PR TITLE
fix scrapper to scrape all jobs

### PR DESCRIPTION
**Pull request by SamMrach**
#5 
@akramaznakour now it scrapes all available jobs. let me what you think of it.

### CODE REVIEW

**Comment by SamMrach** [comment]
@akramaznakour  Hi akram do u have time to check my PR ?

File: content/linkedin-scraper.js [hunk 1]
```diff
@@ -96,14 +96,14 @@ async function scrapeLinkedInJobs() {
       ".jobs-search-results-list"
     );
 
-    await simulateRealScrollToEnd(cardsListElement, 500);
-
-    await new Promise((resolve) => setTimeout(resolve, 500));
+    //await simulateRealScrollToEnd(cardsListElement, 500);
+    await scrollProgressively();
+    //await new Promise((resolve) => setTimeout(resolve, 500));
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:101]: can we remove these comments

File: content/linkedin-scraper.js [hunk 2]
```diff
@@ -96,14 +96,14 @@ async function scrapeLinkedInJobs() {
...
 
     const cards = document.querySelectorAll(".job-card-container");
 
     const cardCount = cards.length;
-
+    console.log("card length :" +cardCount);
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:106]: can you remove the logs?

File: content/linkedin-scraper.js [hunk 3]
```diff
@@ -135,3 +135,20 @@ async function scrapeLinkedInJobs() {
...
+  function scrollProgressively(){
+    return new Promise((res)=>{
+      let jobs=document.querySelectorAll(".job-card-container");
+      let jobsCount=jobs.length;
+      jobs[jobsCount-1].scrollIntoView({behaviour:"smooth"});
+      setTimeout(async() => {
+        if(document.querySelectorAll(".job-card-container").length !=jobsCount)
+        res(ScrollProgressive());
+        else {
+        //console.log("length :"+document.querySelectorAll(".job-card-container").length);
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:148]: can remove this comment

File: content/linkedin-scraper.js [hunk 4]
```diff
@@ -135,3 +135,20 @@ async function scrapeLinkedInJobs() {
     }
   }
 }
+
+  function scrollProgressively(){
+    return new Promise((res)=>{
+      let jobs=document.querySelectorAll(".job-card-container");
+      let jobsCount=jobs.length;
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:142]: lets not use mutable let variable, const only for safety

File: content/linkedin-scraper.js [hunk 5]
```diff
@@ -135,3 +135,20 @@ async function scrapeLinkedInJobs() {
...
+      jobs[jobsCount-1].scrollIntoView({behaviour:"smooth"});
+      setTimeout(async() => {
+        if(document.querySelectorAll(".job-card-container").length !=jobsCount)
+        res(ScrollProgressive());
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:146]: wrong function name

File: content/linkedin-scraper.js [hunk 6]
```diff
@@ -135,3 +135,20 @@ async function scrapeLinkedInJobs() {
...
+        else {
+        //console.log("length :"+document.querySelectorAll(".job-card-container").length);
+        return res("done");
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:149]: Using just "done" might be ambiguous if someone else reads your code or if you revisit it in the future. Adding a little more context to the resolution value can help prevent misunderstandings 

`resolve({ status: "completed" });
`

File: content/linkedin-scraper.js [hunk 7]
```diff
@@ -135,3 +135,20 @@ async function scrapeLinkedInJobs() {
...
+        }
+      }, 2000);
```
Comment (akramaznakour) [inline content/linkedin-scraper.js:151]: magic number

**Review by akramaznakour — COMMENTED** [review]
The function can be improved more 
ChatGPT proposed the following :smiley: :

```JS
async function scrollProgressively(targetSelector, scrollOptions, timeout) {
    return new Promise((resolve, reject) => {
        const targetElements = document.querySelectorAll(targetSelector);
        const initialElementCount = targetElements.length;

        if (initialElementCount === 0) {
            return reject({ success: false, error: "No target elements found." });
        }

        const lastTargetElement = targetElements[initialElementCount - 1];
        lastTargetElement.scrollIntoView(scrollOptions);

        setTimeout(() => {
            const updatedElementCount = document.querySelectorAll(targetSelector).length;

            if (updatedElementCount !== initialElementCount) {
                resolve(scrollProgressively(targetSelector, scrollOptions, timeout));
            } else {
                resolve({ success: true });
            }
        }, timeout);
    });
}

```

**Comment by SamMrach** [comment]
@akramaznakour i made all  proposed changes. let me know your thoughts.

**Review by akramaznakour — APPROVED** [review]

**Comment by akramaznakour** [comment]
Hi, @SamMrach I have approved the PR :smile:  thank you for your contribution :pray:

**Comment by akramaznakour** [comment]
You can squash merge your changes

**Comment by SamMrach** [comment]
> You can squash merge your changes

@akramaznakour  i merged in the main branch of my repo, is it ok ?

**Comment by akramaznakour** [comment]
yes

> > You can squash merge your changes
> 
> @akramaznakour i merged in the main branch of my repo, is it ok ?

Yes, it is okay, you can merge this PR.

**Comment by akramaznakour** [comment]
> @akramaznakour how to merge it ?

Sorry @SamMrach 😅 I think merge is enabled only for the owner, I will fix the permissions for this repository later, I am going to merge your PR

**Pull request merged**